plutus-script-utils/src/Plutus/V1/Utils/Typed/Scripts/Validators.hs

```diff
@@ -26,20 +26,21 @@ module Ledger.Typed.Scripts.Validators
     ) where
 
 import Data.Aeson (FromJSON, ToJSON)
-import Data.Kind
-import Data.Void
+import Data.Kind (Type)
+import Data.Void (Void)
 import GHC.Generics (Generic)
 
 import PlutusCore.Default (DefaultUni)
-import PlutusTx
+import PlutusTx (BuiltinData, CompiledCode, Lift, UnsafeFromData (unsafeFromBuiltinData), applyCode, liftCode)
 import PlutusTx.Prelude (check)
```

Comment:
Can't comment at point, but IMHO the `wrapValidator` function has nothing to do with the `class ValidatorTypes`, `TypedValidator` nor `mkTypedValidator`.

I must say that, now as we have moved away from this "typed interface", our code became clearer and simpler. Maybe it only appears "better" to us now, because we have not been relying on any additional type safety of the associated types.

No action required here though. Maybe: we can find a better name than `wrapValidator` and `WrappedValidatorType`, now that we are breaking interfaces all over the place? The latter is more a `ValidatorType` or `UntypedValidatorType` to me (as a user).

Reply:
Very good point. The typed interface is mostly used by the `plutus-ledger-constraints` (creating transaction using a constraint-based API) library and the StateMachine. If you're not using those, then `ValidatorTypes` is not useful, unless you really care about type safety.

Here's what I can do:

* I'll rename `WrappedValidatorType` to UntypedValidator`.
* I'll rename `wrapValidator` to `toUntypedValidator`. 
* I'll postpone moving `TypedValidator` and related functions in `plutus-script-utils` until I have a clearer idea on how to organize it.

Did I miss anything?

Reply:
I like `UntypedValidator` for the type alias and I suppose `UntypedMonetaryPolicyScript` is the equivalent.

As for the function: None of these sound significantly easier to understand to me :sweat_smile: How do you like just `untypedValidator :: datum -> redeemer -> context -> UntypedValidator` or `mkUntypedValidator`?

Something I realized when vendoring: Making `context` also a type variable with `UnsafeFromData context` allows the same function to be used in multiple plutus versions!

Reply:
> Something I realized when vendoring: Making context also a type variable with UnsafeFromData context allows the same function to be used in multiple plutus versions

I was actually also thinking about that, although I'd rather make it something like
```
type family PlutusContextType LedgerPlutusVersion where
  PlutusContextType PlutusV1 = V1.Context
  PlutusContextType PlutusV2 = V2.Context
```
and then you index by `LedgerPlutusVersion` in some other places. Just letting `context` be anything you can convert from data loses you some safety.